docs/content/routing/routers/index.md

```diff
@@ -0,0 +1,298 @@
+
+### TLS
+
+When specifying a TLS section, you tell Traefik that the current router is dedicated to HTTPS requests only (and that the router should ignore HTTP (non tls) requests).
+Traefik will terminate the SSL connections (meaning that it will send decrypted data to the services).
+
+??? example "Configuring the router to accept HTTPS requests only"
+
+    ```toml
+    [http.routers]
+       [http.routers.Router-1]
+          rule = "Host(`foo-domain`) && Path(`/foo-path/`)"
+          service = "service-id"
+          [http.routers.Router-1.tls] # will terminate the TLS request
+    ```
+
+!!! note "HTTPS & ACME"
+
+    In the current version, with [ACME](../../https-tls/acme.md) enabled, automatic certificate generation will apply to every router declaring a TLS section.
+    In the near future, options will be available to enable fine-grain control of the TLS parameters.
+
+!!! note "Passthrough"
+
+    On TCP routers, you can configure a passthrough option so that Traefik doesn't terminate the TLS connection.
+
+!!! important "Routers for HTTP & HTTPS"
+
+    If you need to define the same route for both HTTP and HTTPS requests, you will need to define two different routers: one with the tls section, one without.
+
+    ??? example "HTTP & HTTPS routes"
+
+        ```toml
+        [http.routers]
+           [http.routers.Router-1-https]
+              rule = "Host(`foo-domain`) && Path(`/foo-path/`)"
+              service = "service-id"
+              [http.routers.Router-1.tls] # will terminate the TLS request
+
+           [http.routers.Router-1-http]
+              rule = "Host(`foo-domain`) && Path(`/foo-path/`)"
+              service = "service-id"
+        ```
+
+## Configuring TCP Routers
+
+### General
+
+If both HTTP routers and TCP routers listen to the same entrypoints, the TCP routers will apply *before* the HTTP routers.
+If no matching route is found for the TCP routers, then the HTTP routers will take over.
+
+### EntryPoints
+
+If not specified, TCP routers will accept requests from all defined entrypoints.
+If you want to limit the router scope to a set of entrypoints, set the entrypoints option.
+
+??? example "Listens to Every EntryPoint"
+
+    ```toml
+    [entrypoints]
+       [entrypoints.web]
+          # ...
+       [entrypoints.web-secure]
+          # ...
+       [entrypoints.other]
+          # ...
+
+    [tcp.routers]
+       [tcp.routers.Router-1]
+          # By default, routers listen to every entrypoints
+          rule = "Host(traefik.io)"
+          service = "service-1"
+    ```
+
+??? example "Listens to Specific EntryPoints"
+
+    ```toml
+    [entrypoints]
+       [entrypoints.web]
+          # ...
+       [entrypoint.web-secure]
+          # ...
+       [entrypoint.other]
+          # ...
+
+    [tcp.routers]
+       [tcp.routers.Router-1]
+          entryPoints = ["web-secure", "other"] # won't listen to entrypoint web
+          rule = "Host(traefik.io)"
```

Comment:
I was wondering if it was not supposed to be:
```
rule = "HostSNI(`traefik.io`)"
```

Reply:
You are correct! Nice catch 👍 
Would you like to PR the change? or shall I?

Reply:
Please review PR #4592